libraries/AdaptiveExpressions/FunctionUtils.cs

```diff
@@ -1021,7 +1054,7 @@ internal static void ValidateForeach(Expression expression)
         {
             if (expression.Children.Length != 3)
             {
-                throw new Exception($"foreach expect 3 parameters, found {expression.Children.Length}");
+                throw new Exception($"foreach expects 3 parameters, found {expression.Children.Length}");
             }
```

Comment:
Do not throw general exceptions, in this cause, ArgumentException would be a better thing to throw.

Reply:
changed to ArgumentException